### ISSUE TITLE
yaml property files are not considered in lint rules

### ISSUE THREAD

**Issue by Kjames5269** [avioconsulting/mule-linter#114]
With the current implementation, you can not check that .yaml property files exist or contain properties.
[Lines 50-57 of the MuleApplication class
](https://github.com/avioconsulting/mule-linter/blob/master/mule-linter-core/src/main/groovy/com/avioconsulting/mule/linter/model/MuleApplication.groovy#L50-L57)uses a name filter to enforce mule property files must end with .properties when mulesoft allows both .properties and .yaml extensions. 

```groovy
    void loadPropertyFiles() {
        File resourcePath = new File(applicationPath, PROPERTY_PATH)
        if (resourcePath.exists()) {
            resourcePath.traverse(nameFilter: ~/.*.properties/) { file->
                propertyFiles.add(new PropertyFile(file))
            }
        }
    }
```

This affects the following property rules in unintuitive ways. Setting the pattern below to end in .yaml will throw an error that the file does not exist regardless if that is true or not.
```groovy
        PROPERTY_FILE_NAMING {
            environments = ['dev', 'test', 'prod']
            pattern = '${appname}-${env}.properties'
        }
        PROPERTY_FILE_COUNT_MISMATCH {
            environments = ['dev', 'test', 'prod']
            pattern = '${appname}-${env}.properties'
        }
```

**Comment by dileepkumarv-avio** [comment]
@adesjardin @TimGonz-AVIO 
Below are the changes required to implement this rule to support yaml file in linter:
1. Current _**MuleApplication.groovy**_ implementation supports only for loading .property file, this needs to be updated to load mule properties files with .properties and .yaml file extenstion
    ```
 void loadPropertyFiles() {
         File resourcePath = new File(applicationPath, PROPERTY_PATH)
         if (resourcePath.exists()) {
             resourcePath.traverse(nameFilter: ~/.*.properties/) { file->
                 propertyFiles.add(new PropertyFile(file))
             }
         }
    }
 ```
2. Below changes are required in **PropertyFile** implementation to support YAML files:
 	a. Update properties type from _java.util.Properties_ to _java.util.Hashtable_ object, and  update _getProperty_ method implementation to pick property from Hashtable object
 	b. Add implementation to parse YAML file contents using SnakeYAML and populate yaml properties into properties object
3. Add test case for all the existing rules using properties files to validate for yaml properties files

**Comment by dileepkumarv-avio** [comment]
Added support for processing YAML properties in linter rule. This has been implemented in PR [#134 ](https://github.com/avioconsulting/mule-linter/pull/134)
